src/client/primitives/base.py

```diff
@@ -53,11 +56,13 @@ def register(self, type_: type, value: E) -> type:
 
     def generate_subtype(self, type_: type, value: E) -> type:
         if not self.extendable:
-            raise NotImplementedError()
+            raise NotImplementedError("Only extendable VariantRegistry can generate subtype")
 
         generated_type_name = '_Generated' + value.value.title() + type_.__name__
         BaseTolokaObjectMetaclass(generated_type_name, (type_,), {}, spec_value=value)
-
+        logger.error(f'{generated_type_name} class was generated. Probably it is a new functionality on the platform.\n'
```

Comment:
I think it is looks more like a warning than an error (since it is now an expected behavior in some cases)